skyportal/handlers/api/gwdetector.py

```diff
@@ -0,0 +1,234 @@
+          description: Retrieve a gwdetector
+          tags:
+            - gwdetectors
+          parameters:
+            - in: path
+              name: gwdetector_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleGWDetector
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all gwdetectors
+          tags:
+            - gwdetectors
+          parameters:
+            - in: query
+              name: name
+              schema:
+                type: string
+              description: Filter by name (exact match)
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfGWDetectors
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        with self.Session() as session:
+
+            if gwdetector_id is not None:
+                t = session.scalars(
+                    GWDetector.select(session.user_or_token).where(
+                        GWDetector.id == int(gwdetector_id)
+                    )
+                ).first()
+                if t is None:
+                    return self.error(
+                        f"Could not load GW Detector with ID {gwdetector_id}"
+                    )
+                return self.success(data=t)
+
+            tel_name = self.get_query_argument("name", None)
+            stmt = GWDetector.select(session.user_or_token)
+            if tel_name is not None:
+                stmt = stmt.where(GWDetector.name == tel_name)
+
+            data = session.scalars(stmt).all()
+            return self.success(data=data)
+
+    @permissions(['Manage allocations'])
+    def put(self, gwdetector_id):
+        """
+        ---
+        description: Update gwdetector
+        tags:
+          - gwdetectors
+        parameters:
+          - in: path
+            name: gwdetector_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema: GWDetectorNoID
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        with self.Session() as session:
+            t = session.scalars(
+                GWDetector.select(session.user_or_token, mode="update").where(
+                    GWDetector.id == int(gwdetector_id)
+                )
+            ).first()
+            if t is None:
+                return self.error('Invalid GW Detector ID.')
+            data = self.get_json()
```

Comment:
A PUT should let you modify or create if doesn't exist. Maybe you want to PATCH in this case?

Reply:
good point.